File: A18Array11/A5BestTimeBuySell.js
let prices = [7, 1, 5, 3, 6, 4];

function bestTimeToBuySell(prices) {
  let maxPrice = Number.MAX_VALUE;
  let maxProfit = 0;

  for (let i = 0; i < prices.length; i++) {
    if (maxPrice < prices[i]) {
      let profit = prices[i] - maxPrice;
      maxProfit = Math.max(maxProfit, profit);
    } else {
      maxPrice = prices[i];
    }
  }
  console.log("maxProfit", maxProfit);
}

bestTimeToBuySell(prices);

File: A18Array11/A5BestTimeBuySell.py
prices = [7, 1, 5, 3, 6, 4]

def bestTimeToBuySell(prices):
    max_price = float('inf')
    max_profit = 0
    
    for price in prices:
        print(price)
        if max_price < price:
            profit = price - max_price
            max_profit = max(max_profit,profit)
        else:
            max_price = price
    print("maxProfit",max_profit)

bestTimeToBuySell(prices);